virtual_machines_type.go

```diff
@@ -245,17 +245,15 @@ type ChangeServiceForVirtualMachineResponse VirtualMachineResponse
 // ResetPasswordForVirtualMachine (Async) represents the scaling of a VM
 //
 // CloudStack API: https://cloudstack.apache.org/api/apidocs-4.10/apis/resetPasswordForVirtualMachine.html
-type ResetPasswordForVirtualMachine ScaleVirtualMachine
+type ResetPasswordForVirtualMachine RebootVirtualMachine
```

Comment:
The names don't seem to fit together. Something automated?

Reply:
They fit, this part is automagically-manually-automated. The doc is buggy though... :-)

```
      "description": "Resets the password for virtual machine. The virtual machine must be in a \"Stopped\" state and the template must alrea>      "isasync": true, 
      "name": "resetPasswordForVirtualMachine",                                                                                               
      "params": [
        {
          "description": "The ID of the virtual machine",
          "length": 255, 
          "name": "id",
          "related": "revertToVMSnapshot,migrateVirtualMachineWithVolume,revertToVMSnapshot,removeNicFromVirtualMachine,startVirtualMachine,r>
          "required": true,
          "type": "uuid"
        }
      ],

      "description": "Reboots a virtual machine.",
      "isasync": true,
      "name": "rebootVirtualMachine",                                                                                                         
      "params": [
        {
          "description": "The ID of the virtual machine",
          "length": 255,
          "name": "id",
          "related": "migrateVirtualMachine,revertToVMSnapshot,migrateVirtualMachineWithVolume,revertToVMSnapshot,removeNicFromVirtualMachine>
          "required": true,
          "type": "uuid"
        }
      ]
```